src/seedu/addressbook/data/person/Name.java

```diff
@@ -36,6 +36,20 @@ public Name(String name) throws IllegalValueException {
     public static boolean isValidName(String test) {
         return test.matches(NAME_VALIDATION_REGEX);
     }
+    
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if they have the same value
+     * after trimming the spaces and converting to lower case
+     */
+     public boolean isSimilar(Name other) {
+    	 if (other == null) {
+    		 return false;
+    	 }
+    	 String thisName = fullName.trim().toLowerCase();
+    	 String otherName = other.fullName.trim().toLowerCase();
+    	 return thisName.equals(otherName);
+     }
```

Comment:
Good try, but you are a bit off from TDD. For example, a TDD session starts by adding a failing test, not by adding an empty method. Please take a look at the course handouts and suggested readings given in the assignment description to get a good understanding of TDD.